raft/server.cc

```diff
@@ -1055,7 +1065,16 @@ future<> server_impl::applier_fiber() {
                // (i.e. didn't yet receive from _apply_entries queue) but will soon. We avoid unnecessary work
                // of taking snapshots ourselves but comparing our last index directly with what's currently in _fsm.
                auto last_snap_idx = _fsm->log_last_snapshot_idx();
-               if (_applied_idx >= last_snap_idx && _applied_idx - last_snap_idx >= _config.snapshot_threshold) {
+
+               // Having additional room of max_command_memory_usage() ensures
```

Comment:
It seems that previously we could get stuck in the same vein:
```
log=0 1 2 3
last_snap_idx = 2
_applied_idx = 3
snapshot_threshold = 2
first_idx = 1
max_log_size = 3
```
In this case we wouldn't have entered the if in `apply_fiber` and skipped making a snapshot, but the next `add_entry' would have hung on the semaphore forever.